core/src/main/java/org/apache/calcite/sql/validate/SqlValidatorImpl.java

```diff
@@ -646,8 +646,8 @@ private boolean addOrExpandField(List<SqlNode> selectItems, Set<String> aliases,
   }
 
   public SqlNode validate(SqlNode topNode) {
-    SqlValidatorScope scope = new EmptyScope(this);
-    scope = new CatalogScope(scope, ImmutableList.of("CATALOG"));
+    SqlValidatorScope scope = new CatalogScope(new EmptyScope(this),
+            ImmutableList.of("CATALOG"));
```

Comment:
What's the different between and after your patch, it seems the logic are equivalent.

Reply:
I also think so.

Reply:
yes, I mistake.